flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java

```diff
@@ -176,14 +176,17 @@ public static SavepointRestoreSettings forPath(
     public static void toConfiguration(
             final SavepointRestoreSettings savepointRestoreSettings,
             final Configuration configuration) {
-        configuration.set(
-                StateRecoveryOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE,
-                savepointRestoreSettings.allowNonRestoredState());
-        configuration.set(
-                StateRecoveryOptions.RESTORE_MODE, savepointRestoreSettings.getRecoveryClaimMode());
-        final String savepointPath = savepointRestoreSettings.getRestorePath();
-        if (savepointPath != null) {
-            configuration.set(StateRecoveryOptions.SAVEPOINT_PATH, savepointPath);
+        if (!savepointRestoreSettings.equals(SavepointRestoreSettings.none())) {
```

Comment:
I am curious how this fix works. I see that the only case the fix will effect is the none case , how does the unit test relate to the none case?